### ISSUE TITLE
azurerm_windows_virtual_machine  must be replaced - but does not indicate why

### ISSUE THREAD

**Issue by Marcus-James-Adams** [terraform-providers/terraform-provider-azurerm#11558]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

Terraform v0.15.1
on windows_amd64

### Affected Resource(s)

* `azurerm_windows_virtual_machine`

### Terraform Configuration Files

```hcl
# azurerm_windows_virtual_machine.server must be replaced
-/+ resource "azurerm_windows_virtual_machine" "server" {
      # Warning: this attribute value will be marked as sensitive and will
      # not display in UI output after applying this change
      ~ admin_password             = (sensitive value)
      ~ id                         = "/subscriptions/4047c6c4-f7cd-423a-81cb-XXXXX/resourceGroups/server-1-rsg/providers/Microsoft.Compute/virtualMachines/server-1" -> (known after apply)
        name                       = "server-1"
      ~ private_ip_address         = "10.100.120.4" -> (known after apply)
      ~ private_ip_addresses       = [
          - "10.100.120.4",
        ] -> (known after apply)
      + public_ip_address          = (known after apply)
      ~ public_ip_addresses        = [] -> (known after apply)
      ~ tags                       = {
          ~ "cost_center"           = "N/A" -> "AzureGen"
          + "shutdown-ooh"          = "false"
          + "shutdown-paused"       = "false"
          - "shutdown_ooh"          = "false" -> null
          - "shutdown_paused"       = "false" -> null
          ~ "tf_code_last_modified" = "30th June 2020" -> "2nd May 2021"
          ~ "tf_code_version"       = "1.7" -> "1.8"
            # (10 unchanged elements hidden)
        }
      ~ virtual_machine_id         = "3d1fb4ce-03d3-4782-9044-d7edc78747c8" -> (known after apply)
      + zone                       = (known after apply)
        # (18 unchanged attributes hidden)


      ~ additional_unattend_content {
          # Warning: this attribute value will be marked as sensitive and will
          # not display in UI output after applying this change
          ~ content = (sensitive value)
            # (1 unchanged attribute hidden)
        }
      ~ additional_unattend_content {
          ~ content = (sensitive value)
            # (1 unchanged attribute hidden)
        }


      ~ os_disk {
          ~ name                      = "server-1_OsDisk_1_91ed859672e24e14a31dacc000e2e5d9" -> (known after apply)
            # (4 unchanged attributes hidden)
        }
        # (2 unchanged blocks hidden)
    }

```

### Expected Behaviour

Should show only tags have changed and update the VM

### Actual Behaviour

Says the VM must be replaced but does not say why

### Steps to Reproduce

1. `terraform apply`

### References

* #0000

**Comment by magodo** [comment]
@Marcus-James-Adams This is probably fixed by hashicorp/terraform#28583 since Terraform v0.15.2.

**Comment by katbyte** [comment]
closing as @magodo indicated this has been fixed a number of months ago
